feat: upstream parameters on oidc (#1307)
ory/docs: docs/kratos/bring-your-own-ui/10_custom-ui-advanced-integration.mdx

</form>
```

#### Upstream provider parameters

> In some cases, it is possible to pass additional parameters to the upstream provider. For example, when using the Googleprovider,

Reviewer: ```suggestion
In some cases, it is possible to pass additional parameters to the upstream provider. For example, when using the Google provider,
```

</form>
```

#### Upstream provider parameters

> In some cases, it is necessary to pass additional parameters to the upstream provider. For example, when using the Google

Reviewer: ```suggestion
In some cases, it is possible to pass additional parameters to the upstream provider. For example, when using the Google
```
Presumably, passing these Google parameters is optional? In which case I think it's clearer to say that it is "possible" rather than "necessary" to pass these parameters.

provider, it is possible to pass the `login_hint` and `hd` parameters.

The `login_hint` parameter suppresses the account chooser and either pre-fills the email box on the sign-in form, or selects the

> proper session. The `hd` parameter limits the login/registration process to a Google Organization, e.g. `mycollege.edu`.

Reviewer: ```suggestion
proper session. The `hd` parameter limits the login/registration process to a Google Organization, for example `mycollege.edu`.
```

To pass these parameters on `login`, `registration` and `settings` flows, the application can use the submit form body to pass the
parameters. For example:

```html
<form action="https://{project.slug}.projects.oryapis.com/self-service/login?flow=<flow-id>" method="POST">
  <input type="submit" name="provider" value="google" />
  <input type="hidden" name="upstream_parameters.login_hint" value="foo@bar.com" />
  <input type="hidden" name="upstream_parameters.hd" value="bar.com" />
</form>
```

:::note

> Only the `login_hint` and `hd` parameters are currently supported. More parameters can be supported in the future.

Reviewer: ```suggestion
Only the `login_hint` and `hd` parameters are currently supported.
```
